test/common/Client.js

```diff
@@ -1634,6 +1634,57 @@ describe('The Client class', function() {
       });
     });
 
+    it('should repeat authentication steps if server demands to', function(done) {
+      var authAccessResponse = {
+        accessToken: 'accessToken1',
+        api: '/',
+        eventSource: '/es',
+        upload: '/upload',
+        download: '/download'
+      };
+
+      new jmap.Client({
+        post: function(url, headers, data) {
+          if (data.username) {
+            return q({
+              continuationToken: 'continuationToken1',
+              methods: ['password']
+            });
+          } else if (data.password === 'pwd1') {
+            return q({
+              continuationToken: 'continuationToken2',
+              prompt: '2nd Auth Factor',
+              methods: ['password']
+            });
+          } else if (data.password === 'pwd2') {
+            return q({
+              continuationToken: 'continuationToken3',
+              prompt: '3rd Auth Factor',
+              methods: ['password']
+            });
+          } else {
+            return q(authAccessResponse);
+          }
+        }
+      })
+      .withAuthenticationUrl('https://test')
+      .authenticate('user@domain.com', 'Device name', function(authContinuation) {
+        if (authContinuation.continuationToken === 'continuationToken1') {
+          return q({ method: 'password', password: 'pwd1' });
+        } else if (authContinuation.continuationToken === 'continuationToken2') {
+          return q({ method: 'password', password: 'pwd2' });
+        } else {
+          return q({ method: 'password', password: 'pwd3' });
+        }
+      })
+      .then(function(authAccess) {
+        expect(authAccess).to.deep.equal(authAccessResponse);
+        done();
+      }, function(err) {
```

Comment:
We usually directly give the `done` function as the error callback in such case.